tests/unit/deploy/test_deployer.py

```diff
@@ -149,18 +149,71 @@ def test_policy_autogenerated_when_enabled(app_policy,
     assert 'Version' in generated
 
 
+def test_can_load_non_stage_specific_name(app_policy, in_memory_osutils):
+    # This is a test for backcompat loading of .chalice/policy.json
+    # for the dev stage.  The default name is suppose to include
+    # the chalice stage name, e.g. policy-dev.json, but to support
+    # existing use cases we'll look for .chalice/policy.json only
+    # if you're in dev stage.
+    previous_policy = '{"Statement": ["foo"]}'
+    filename = os.path.join('.', '.chalice', 'policy.json')
+    in_memory_osutils.filemap[filename] = previous_policy
+    config = Config.create(project_dir='.', autogen_policy=False)
+    generated = app_policy.generate_policy_from_app_source(config)
+    assert generated == json.loads(previous_policy)
+
+
+def test_legacy_file_not_loaded_in_non_dev_stage(app_policy, in_memory_osutils):
+    previous_policy = '{"Statement": ["foo"]}'
+    filename = os.path.join('.', '.chalice', 'policy.json')
+    in_memory_osutils.filemap[filename] = previous_policy
+    config = Config.create(project_dir='.', autogen_policy=False,
+                           chalice_stage='not-dev')
+    generated = app_policy.generate_policy_from_app_source(config)
+    # We should not have loaded the previously policy from policy.json
+    # because that's only supported for the 'dev' stage.
+    assert generated != json.loads(previous_policy)
+
+
+def test_can_provide_stage_specific_policy_file(app_policy, in_memory_osutils):
+    policy_filename = 'my-custom-policy.json'
+    config = Config.create(project_dir='.', autogen_policy=False,
+                           iam_policy_file=policy_filename,
+                           chalice_stage='dev')
+
+    previous_policy = '{"Statement": ["foo"]}'
+    filename = os.path.join('.', '.chalice', policy_filename)
+    in_memory_osutils.filemap[filename] = previous_policy
+    generated = app_policy.generate_policy_from_app_source(config)
+    assert generated == json.loads(previous_policy)
+
+
+def test_can_provide_stage_specific_policy_for_other_stage(app_policy,
```

Comment:
Is there a test for when ``autogen_policy`` is ``True``, but the stage is not dev? Looks like you have one for when the stage is dev, but not when it is the non default stage.